### PR TITLE
Assorted C++ build fixes for review

### PR DESCRIPTION
**Pull request by ssorj**
Do not merge.  For review purposes only.

This includes the sketchy initialization changes to make things compile on Fedora 26.  It now also includes the clang linking fix, eliminates the swig cmake warnings, and bumps the max tested proton version.

### CODE REVIEW

**Comment by chrisrichardson77** [comment]
I previous wrote some rather verbose comments about the initialisation warning, then deleted them thinking I was perhaps stating the obvious. Since there hasn't really been any other input I feel I should re-instate them but I'll try to keep it short this time!

File: src/qpid/broker/Selector.cpp [hunk 1]
```diff
@@ -108,7 +108,7 @@ MessageSelectorEnv::MessageSelectorEnv(const Message& m) :
 
 const Value MessageSelectorEnv::specialValue(const string& id) const
 {
-    Value v;
+    Value v = Value();
```
Comment (chrisrichardson77) [inline src/qpid/broker/Selector.cpp]: The original version of this is correct and invokes the default constructor.
The new version is incorrect - it default-constructs a temporary which is copied/moved/elided (depending on compiler version and standard) to v.

File: src/qpid/broker/Selector.cpp [hunk 2]
```diff
@@ -165,8 +165,6 @@ const Value MessageSelectorEnv::specialValue(const string& id) const
             returnedStrings.push_back(new string(jmsType));
             v = returnedStrings[returnedStrings.size()-1];
         }
-    } else {
-        v = Value();
```
Comment (chrisrichardson77) [inline src/qpid/broker/Selector.cpp]: Correct (for the record), should be removed irrespective of above initialisation

File: src/qpid/broker/SelectorExpression.cpp [hunk 3]
```diff
@@ -1036,7 +1036,7 @@ Expression* unaryArithExpression(Tokeniser& tokeniser)
 Expression* parseExactNumeric(const Token& token, bool negate)
 {
     int base = 0;
-    string s;
+    string s("");
```
Comment (chrisrichardson77) [inline src/qpid/broker/SelectorExpression.cpp]: Not necessary as std::string's default constructor already assigns an empty string.

File: src/qpid/broker/SelectorToken.h [hunk 4]
```diff
@@ -69,7 +69,8 @@ struct Token {
     std::string val;
     std::string::const_iterator tokenStart;
 
-    Token()
+    Token() :
+        type()
```
Comment (chrisrichardson77) [inline src/qpid/broker/SelectorToken.h]: The base type of an enum is a primitive integral type so it is correct to initialise it here. However "type()" will assign zero to the variable irrespective of whether that is a valid enum element - perhaps better to explicitly assign an enum element eg: type(T_EOS) (if that was the intention) or type(T_NULL).

**Comment by ssorj** [comment]
@chrisrichardson77, thanks for taking a look at this.  I really appreciate it.

I adjusted my changes locally in according to your suggestions.  Only one initialization problem arises now.

  https://gist.github.com/ssorj/cb3da19880716089becdb7ff95837356

That does indeed look like a specious GCC [maybe-]warning.  It looks like other people have run into this.

  https://stackoverflow.com/questions/21755206/how-to-get-around-gcc-void-b-4-may-be-used-uninitialized-in-this-funct

I currently see two options.  I can keep the unnecessary s("") to avoid this particular warning, or we can deploy the big hammer and stop failing the build on maybe-uninitialized warnings.

**Comment by chrisrichardson77** [comment]
There is a third option as mentioned in the stackoverflow article you linked - that of #pragma directives <shudder>... arguably even more ugly than the 's("")' syntax so we might have to live with 's("")' to get around gcc's idiosyncrasies. I'm no lover of big hammers so 's("")' would get my vote as the least of the available evils.

**Pull request merged**